apm-agent-plugins/apm-urlconnection-plugin/src/main/java/co/elastic/apm/agent/urlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -64,7 +65,7 @@
 
     @Override
     public ElementMatcher<? super TypeDescription> getTypeMatcher() {
-        return hasSuperType(is(HttpURLConnection.class));
+        return hasSuperType(is(HttpURLConnection.class)).and(not(named("sun.net.www.protocol.https.HttpsURLConnectionImpl")));
```

Comment:
```suggestion
        return hasSuperType(is(HttpURLConnection.class))
            .and(not(named("sun.net.www.protocol.https.HttpsURLConnectionImpl")))
            .and(declaresField(named("connected")));
```

Reply:
Why `declaresField(named("connected"))`? It can be an inherited and not a declared field.

Reply:
@felixbarny This cannot work, as the `connected` field belongs to `java.net.URLConnection` (abstract). Let me know if you think we can merge it.

Reply:
Yes, forget what I said. `@Advice.FieldValue("connected") boolean connected` is guaranteed to work as the `connected` field is always accessible as we're searching for sub-types of HttpUrlConnection that are also sub-types of URLConnection